driver/ssh.go

```diff
@@ -91,20 +96,29 @@ func (d *SSH) RunCommand(command string) (string, error) {
 	return string(out), nil
 }
 
-func (d *SSH) GetDetails() string {
-	return fmt.Sprintf(`SSH - %s`, d.String())
-}
-
-func NewSSHForTest() *SSH {
-	return &SSH{
-		User:            "dev",
-		Host:            "127.0.0.1",
-		Port:            2222,
-		KeyFile:         "/home/deven/.ssh/id_rsa",
-		KeyPass:         "",
-		CheckKnownHosts: false,
-		fields: fields{
-			PollInterval: 5,
-		},
+func (d *SSH) GetDetails() SystemDetails {
+	if d.Info == nil {
+		uname, err := d.RunCommand(`uname`)
+		// try windows command
+		if err != nil {
```

Comment:
Probably need to check the error could be other errors. Like ssh connection errors.

Reply:
And then what do we do at that point